.github/workflows/build-and-test.yaml

```diff
@@ -0,0 +1,58 @@
+name: Build/Test
+
+on:
+  workflow_call:
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3 # v3
+      - name: Install dependencies
+        run: python3 -m pip install tox
+      - name: Run linters
+        run: tox -e lint
+
+  unit-test:
+    name: Unit tests
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3 # v3
+      - name: Install dependencies
+        run: python -m pip install tox
+      - name: Run tests
+        run: tox -e unit
+
+  integration-test:
+    name: Integration tests
+    runs-on: ubuntu-20.04
```

Comment:
Why 20.04? Lets use 22.04 if there is nothing speaking against it.

Reply:
the charm supports building on 20.04 to support running on 20.04 as well. Hence had to run the integration tests on 20.04.